src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -278,7 +278,7 @@ private boolean runMainClass(Path jarPath)
             final Object jarLauncher = jarLauncherConstructor.newInstance(jarFileArchive);
             checkEnvironmentProperties();
             args = parseArguments(args);
-            launchMethod.invoke(jarLauncher, new Object[]{args});
+            launchMethod.invoke(jarLauncher, (Object[]) new Object[]{args});
```

Comment:
With the original new `Object[]{args}`, Sonar complains about the ambiguity which is fair but with this new change Sonar complains about the redundant cast to Object[]. 

All Sonar Lint complaints are resolved when I try: 
`            launchMethod.invoke(jarLauncher, (Object[]) args);
`

Reply:
The reflection fails if we do just cast.
We can suppress the Sonar complaint on redundant cast or the original complaint.
`launchMethod.invoke(jarLauncher, (Object[]) new Object[]{args});`

Reply:
Ok 👍  Please change your target branch to master instead of 8.10.z and I can approve

Reply:
This fix is needed in 8.10 and IDETECT-3817 will be retested by QA.